Classes/AlgorandSecureBackup/AlgorandSecureBackupFlowCoordinator.swift

```diff
@@ -120,7 +120,15 @@ extension AlgorandSecureBackupFlowCoordinator {
 
     private func dismissScreen(from: UIViewController) {
         if launchTransition == .push {
-            from.navigationController?.setViewControllers([presentingScreen], animated: true)
+            guard let navController = from.navigationController else {
+                from.dismissScreen()
+                return
+            }
+            if let settingsVC = navController.viewControllers.first(where: { $0 === presentingScreen }) {
+                navController.popToViewController(settingsVC, animated: true)
+            } else {
+                navController.setViewControllers([presentingScreen], animated: true)
+            }
```

Comment:
Consider using `contains` here.
```suggestion
            if navController.viewControllers.contains(presentingScreen) {
                navController.popToViewController(presentingScreen, animated: true)
            } else {
                navController.setViewControllers([presentingScreen], animated: true)
            }
```